### PR TITLE
Minor bug fixes

### PR DESCRIPTION
**Pull request by arrjon**
Some minor bug fixes. These should be corrected in the main version too.

### CODE REVIEW

**Comment by stephanmg** [comment]
@arrjon not sure what's the issue, thanks for spending time to debug this.

**Comment by arrjon** [comment]
I changed the oldest python version from python 3.9 to 3.10 and updated the GitHub continous integration workflow. This helped with most of the failing tests.

**Comment by arrjon** [comment]
In the migration test, incompatible `numpy` and `pandas` versions caused an error. This is fixed now.

**Comment by arrjon** [comment]
@stephanmg you can review now: only the tests for python 3.9 are sill in the status "expected" and never leave it. Not quite sure why, but we removed python 3.9 anyway.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -3,14 +3,14 @@ name: Deploy
...
-   
+
 
 jobs:
   deploy:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.9]
+        python-version: [3.11]
```
Comment (stephanmg) [inline .github/workflows/deploy.yml:13]: Not  sure if we need to specify python-version here for the deploy workflow. would either go with 3.12 or use the ubuntu-latest default?

**Review by stephanmg — APPROVED** [review]
Only minor comments, fixes look good to me. Thanks for the PR.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -7,23 +7,23 @@ on:
     - main
   pull_request:
   schedule:
-    # run Monday and Thursday at 03:18 UTC
```
[the remark this answers is not in the document]
Comment (arrjon) [reply .github/workflows/ci.yml:10]: actually it is correct, see: https://crontab.cronhub.io

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (arrjon) [reply .github/workflows/deploy.yml]: yes, we need that. the python version is later referenced

File: .github/workflows/ci.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (stephanmg) [reply .github/workflows/ci.yml:10]: Ah, true, sorry ;-) then fine obv

**Pull request merged**